### PR TITLE
populate

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

**Review by MoranOshiaMoveo — APPROVED** [review]

File: src/controllers/dishController.ts [hunk 1]
```diff
@@ -14,17 +11,27 @@ export const getAllDishes = async (
...
-  req: express.Request,
-  res: express.Response
-) => {
+export const getDishByIngredient = async (req: express.Request, res: express.Response) => {
+  try {
+    const { ingredient } = req.params;
+    const dishes = await DishModel.find();
+    const dishesWithIngredient = dishes.filter((dish) => {
+      const ingredients = dish.ingredients.split(",").map((dish) => dish.trim().toLowerCase());
+      return ingredients.includes(ingredient.toLowerCase());
```
Comment (MoranOshiaMoveo) [inline src/controllers/dishController.ts:20]: u'll need to lower case the ingredient that u're getting from the params if the user is typing with big letters

File: src/controllers/generalController.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import express from "express";
+import { DishModel } from "../models/dish";
+import { RestaurantModel } from "../models/restaurant";
+import { ChefModel } from "../models/chef";
+
+export const getResultsBySearch = async (req: express.Request, res: express.Response) => {
```
Comment (MoranOshiaMoveo) [inline src/controllers/generalController.ts]: if we're using post api call the terminology of get is not correct, try to change it to searchByInput or similar

File: src/routes/generalRoute.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import express from "express";
+
+import { getResultsBySearch } from "../controllers/generalController";
+
+export default (router: express.Router) => {
+  router.get("/search", getResultsBySearch);
```
Comment (MoranOshiaMoveo) [inline src/routes/generalRoute.ts]: if we're have a request body' it's more correct to use post and not get method

File: src/controllers/generalController.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  try {
+    const { searchTerm } = req.body;
+
+    if (!searchTerm) {
+      return res.status(400).json({ message: "Please provide a search term" });
+    }
```
Comment (MoranOshiaMoveo) [inline src/controllers/generalController.ts:12]: same in here u need to set the search input also to lower case

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/controllers/generalController.ts (hunk 2 above)
Comment (shirrotem) [reply src/controllers/generalController.ts]: fixed

File: src/routes/generalRoute.ts (hunk 3 above)
Comment (shirrotem) [reply src/routes/generalRoute.ts]: fixed

File: src/controllers/dishController.ts (hunk 1 above)
Comment (shirrotem) [reply src/controllers/dishController.ts:20]: fixed

File: src/controllers/generalController.ts (hunk 4 above)
Comment (shirrotem) [reply src/controllers/generalController.ts:12]: fixed

**Review by MoranOshiaMoveo — APPROVED** [review]

**Pull request merged**